numba/cuda/compiler.py

```diff
@@ -938,22 +1013,27 @@ def specialized(self):
 
     @property
     def definition(self):
+        warn('Use overloads instead of definition',
+             category=NumbaDeprecationWarning)
```

Comment:
Would be good to add a test to check these warnings fire correctly (numerous new deprecation warnings are being added, will note with `test:deprecation`).

Reply:
Test added in https://github.com/numba/numba/pull/6432/commits/d9534677c20a7ab9b13e1a08df26024ed9438f8c